algorithms/ar-kmp/javascript/knuth_morris_pratt.js

```diff
@@ -0,0 +1,83 @@
+function buildPrefixTable(search) {
+  const prefixTable = Array(search.length);
+  let pos = 1
+  let cnd = 0;
+
+  prefixTable[0] = -1;
+
+  while ( pos < search.length) {
+    
+    if (search[pos] == search[cnd]) {
+      
+      prefixTable[pos] = prefixTable[cnd];
+    
+    } else {
+
+      prefixTable[pos] = cnd;
+      cnd = prefixTable[cnd];
+      
+      while (cnd >= 0 && search[pos] != search[cnd]) {
+        
+        cnd = prefixTable[cnd];
+      
+      }
+
+    }
+
+    pos += 1;
+    cnd += 1;
+  
+  }
+
+  return prefixTable;
+}
+
+function knuthMorrisPratt (phrase, word) {
+  const phraseLength = phrase.length;
+  const wordLength = word.length;
+  const prefixTable = buildPrefixTable(word);
+  const matches = [];
+  let i = 0;
+  let j = 0;
+
+  while (i < phraseLength) {
+    
+    if (phrase[i] == word[j]) {
```

Comment:
Codacy found an issue: [Expected '===' and instead saw '=='.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395128)